h/session.py

```diff
@@ -55,9 +59,12 @@ def _current_groups(request):
 
 
 def includeme(config):
-    config.add_settings({
-        "redis.sessions.cookie_httponly": True,
-        "redis.sessions.cookie_max_age": 2592000,
-        "redis.sessions.timeout": 604800,
-    })
-    config.include('pyramid_redis_sessions')
+    settings = config.registry.settings
+
+    factory = SignedCookieSessionFactory(
+        secret=derive_key(settings['secret_key'], b'h.session.cookie_secret'),
```

Comment:
Maybe just add a comment here that by default `derive_key` generates a 64-byte (512-bit) secret, which is the correct length for a SHA512-based HMAC as specified by `hashalg='sha512'`, but if `hashalg` changes, we may need to derive a longer secret.